### PR TITLE
Fix/ecp 1475 trash ct1

### PR DESCRIPTION
**Pull request by stratease**
[TEC-4744](https://theeventscalendar.atlassian.net/browse/TEC-4744) + [ECP-1475](https://theeventscalendar.atlassian.net/browse/ECP-1475) + [ECP-1480](https://theeventscalendar.atlassian.net/browse/ECP-1480)

Is related to work done in https://github.com/the-events-calendar/events-pro/pull/2231

This handles a number of items regarding event trashing and old event cleanup systems, deprecating our old Event Cleaner delete event systems in lieu of using the built in `EMPTY_TRASH_DAYS` functionality of Wordpress. 

Artifacts:
image

See tests for more.

[TEC-4744]: https://theeventscalendar.atlassian.net/browse/TEC-4744?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ECP-1475]: https://theeventscalendar.atlassian.net/browse/ECP-1475?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ECP-1480]: https://theeventscalendar.atlassian.net/browse/ECP-1480?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Events/Custom_Tables/V1/Events/Event_Cleaner/Provider.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	 * occurrences.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $sql The original query to retrieve expired events.
+	 *
+	 * @return string The modified CT1 query to retrieve expired events.
+	 */
+	public function filter_tribe_events_delete_old_events_sql( string $sql ): string {
+		return tribe( Event_Cleaner::class )->filter_tribe_events_delete_old_events_sql( $sql );
```
Comment (lucatume) [inline src/Events/Custom_Tables/V1/Events/Event_Cleaner/Provider.php:90]: I would take a "Controller" approach here and avoid having a separated class, the `Event_Cleaner` one, that only has 1 method.
I would move the `Event_Clenaer::filter_tribe_events_delete_old_events_sql` method body to this one and remove the `Event_Cleaner` class.
This is personal taste, and not a required taste.

Comment (lucatume) [reply src/Events/Custom_Tables/V1/Events/Event_Cleaner/Provider.php:90]: The idea of naming methods attached to filters and actions, indifferent of the comment above, is to say what they do, not where they operate, when possible.

I would rename `filter_tribe_events_delete_old_events_sql` to `redirect_old_events_sql`, or something along those lines.

File: src/Tribe/Event_Cleaner.php [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?php
-
+// @todo This only handles update option actions? Not clear, perhaps naming/refactor a bit.
```
Comment (lucatume) [inline src/Tribe/Event_Cleaner.php:2]: Either rename it now, or remove this todo.

File: src/Tribe/Event_Cleaner_Scheduler.php [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?php
-
+// @todo This handles the cleaning AND scheduling...
```
Comment (lucatume) [inline src/Tribe/Event_Cleaner_Scheduler.php:2]: Is this a todo or a consideration about the unclear naming of this class? If the second is true, then either rename the class and remove the todo or just remove the todo.

File: tests/ct1_integration/Tribe__Events__Event_Cleaner_SchedulerTest.php [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+		// Should be trashed - results will not contain the updates to the posts.
+		$cleaner = new Tribe__Events__Event_Cleaner_Scheduler();
+		$results = $cleaner->move_old_events_to_trash();
+		$this->assertCount( 1, $results );
+
+		// Now let's force WP's trash cleanup. Default is to delete posts 30 days old trash, let's modify the meta to flag this ready to remove
+		$meta_value = time() - ( ( EMPTY_TRASH_DAYS * DAY_IN_SECONDS ) + 1 );
+		update_post_meta( $post->ID, '_wp_trash_meta_time', $meta_value );
+		wp_scheduled_delete();
```
Comment (lucatume) [inline tests/ct1_integration/Tribe__Events__Event_Cleaner_SchedulerTest.php:109]: Nice!

**Review by lucatume — CHANGES_REQUESTED** [review]
Some minor changes related to naming, "pensive" todos and, possibly, some code cleanup.

File: src/Events/Custom_Tables/V1/Events/Event_Cleaner/Provider.php (hunk 1 above)
Comment (stratease) [reply src/Events/Custom_Tables/V1/Events/Event_Cleaner/Provider.php:90]: >  ... method body to this one ...
To which one?

File: src/Tribe/Event_Cleaner.php (hunk 2 above)
Comment (stratease) [reply src/Tribe/Event_Cleaner.php:2]: Whoops.

File: src/Events/Custom_Tables/V1/Models/Builder.php
Comment (stratease) [inline src/Events/Custom_Tables/V1/Models/Builder.php:1]: FYI @lucatume made some updates here to easily accommodate multiple order by's in the Builder.

File: src/Events/Custom_Tables/V1/Models/Builder.php [hunk 5]
```diff
@@ -926,6 +928,23 @@ public function get() {
 		return $this->create_collection( $results );
 	}
 
+	/**
+	 * Compiles the current order by statements if any exist and returns the entire `ORDER BY` clause.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
```
Comment (lucatume) [inline src/Events/Custom_Tables/V1/Models/Builder.php:936]:
```suggestion
	 * @return string The compiled ORDER BY clause.
```

**Review by lucatume — APPROVED** [review]

**Review by lucatume — APPROVED** [review]

File: src/Events/Custom_Tables/V1/Events/Event_Cleaner/Provider.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+	/**
+	 * Deprecating/removing 'tec.event-cleaner' and the scheduler. This is now being handled by the CT1 Event Cleaner.
+	 * system in CT1.
+	 *
+	 * @since TBD
+	 */
+	public function remove_old_recurrence_cleaners() {
+		// Hide from settings page.
+		add_filter( 'tribe_settings_tab_fields', function ( $args, $id ) {
```
Comment (Camwyn) [inline src/Events/Custom_Tables/V1/Events/Event_Cleaner/Provider.php:56]: You can use `tribe_general_settings_tab_fields` for this:
https://github.com/the-events-calendar/the-events-calendar/blob/master/src/admin-views/tribe-options-general.php#L412

**Review by Camwyn — CHANGES_REQUESTED** [review]
Docblock tweak and a simpler filter

**Comment by stratease** [comment]
@Camwyn good points. Updated!

**Review by Camwyn — APPROVED** [review]

**Comment by Camwyn** [comment]
Reverted manually here: https://github.com/the-events-calendar/the-events-calendar/pull/4204

**Pull request merged**